azure-servicemanagement-legacy/tests/test_legacy_mgmt_misc.py

```diff
@@ -2166,7 +2166,7 @@ def test_list_os_images_public(self):
         self.assertIsNotNone(image.is_premium)
         self.assertIsNotNone(image.icon_uri)
         self.assertIsNotNone(image.privacy_uri)
-        self.assertGreaterEqual(len(image.recommended_vm_size), 0)
+        self.assertIsNotNone(image.recommended_vm_size)
```

Comment:
The difficult part here is that the "list" operation result could change over the time. The test takes one OSImage randomly and tests the attributes, we don't know which OSImage was picked, we don't know if `recommended_vm_size` will be set . And fixing a specific label will fail in time (the image will disappear for sure).
For Travis it's not a problem, the tests are recorded so we are tested the same payload again and again. We could (should...) rerun all the tests "live" sometimes to make sure the RestAPI didn't change, but we don't currently (especially with ASM).

I guess the best way is to check that at least one OSImage has a `recommended_vm_size`, this is unlikely than all OSImages have an empty `recommended_vm_size`.

Thoughts?